src/desktop/version.cpp

```diff
@@ -381,7 +381,7 @@ std::string os_version()
 			}
 	}
 
-	if(v.szCSDVersion && *v.szCSDVersion) {
+	if(*v.szCSDVersion) {
```

Comment:
What's the reason behind this change?

Reply:
Clang called it out as a warning, which strict mode convert to an error:
`address of array 'v.szCSDVersion' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]`

Reply:
The other warnings beside the microsoft one that've been added to the cmake script would also need to have more targeted fixes, either by telling it to ignore the warning for specific lines (usually boost includes) or other minor code changes.

Reply:
Ah, it's an array. Makes sense.

Reply:
I see the PR was closed, but should this change be made anyway?

Reply:
Done in af5ddbbe2ce6e42943e620250072a16d1d6202ba